client/service/src/builder.rs

```diff
@@ -949,6 +1017,30 @@ ServiceBuilder<
 			));
 		}
 
+		// Prometheus exporter and metrics
+		let metrics = if let Some(port) = config.prometheus_port {
+			let registry = match prometheus_registry {
+				Some(registry) => registry,
+				None => Registry::new_custom(Some("substrate".into()), None)?
+			};
+
+			let metrics = ServiceMetrics::register(&registry)?;
+
+			let future = select(
+				prometheus_exporter::init_prometheus(port, registry).boxed(),
+				exit.clone()
+			).map(drop);
+
+			let _ = to_spawn_tx.unbounded_send((
+				Box::pin(future),
+				From::from("prometheus-on-block")
```

Comment:
I am not sure what "prometheus-on-block" relates to. I would expect something like `prometheus-server` or `prometheus-endpoint`. Can you explain your thoughts?

Reply:
I didn't understand a little bit about what this was, so I copied the Telemetry, but I think it's the `prometheus-endpoint`